executive.c

```diff
@@ -51,7 +51,7 @@ fcopy (int in_fd, int out_fd, size_t gap, bool stop_if_input_unlocked)
     return -1;
   /* Optimisation (on Linux it double the readahead window) */
   posix_fadvise (in_fd, (off_t) 0, (off_t) 0, POSIX_FADV_SEQUENTIAL);
-  posix_fadvise (in_fd, (off_t) 0, (off_t) 0, POSIX_FADV_WILLNEED);
+  posix_fadvise (in_fd, (off_t) 0, (off_t) 0, POSIX_FADV_NOREUSE);
```

Comment:
FADV_NOREUSE is ignored https://github.com/torvalds/linux/blob/master/mm/fadvise.c#L115 , WILLNEED on the other hand should speed up the upcoming reads.